rfc/0033-quilt-mappings-and-clean-room.md

```diff
@@ -0,0 +1,97 @@
+
+### Issues With the Cleanroom
+
+While discussing how Quilt's approach to mappings should work, the Quilt Mappings Team has made the following observations:
+
+* Almost everyone that ended up contributing to Fabric and Yarn were either switching from other projects or had minor exposure to them, generally Spigot or Forge, so they had been exposed in great amounts to strictly licensed mappings.
+* Regardless of who contributed a name, the CC0's warranty and infringement disclaimers mean that Quilt would be fully responsible in the event of DMCA on our mappings.
+* Some Yarn contributors have been known to be working with other mappings projects, either in their own mods or directly, while remaining active Yarn contributors - even with video evidence, in some cases. Since Quilt does not have the time or resources to prove each name given by these contributors is unique and original, the option that would be most honest to our users is to declare that some names have uncertain origins, and may carry legal risk.
+* Some Yarn names are lifted directly from Mojmap through the use of string constants. Quilt does not have a license from Mojang to use these constants, so they would fall under the Minecraft EULA, which is stricter than the cc0 and infringes on Mojang's copyright.
+
+For these reasons, the Quilt Mappings Team has concluded that it's impossible to say with absolute certainty that all Yarn names are unique, and not inspired by (or copied from) other mappings projects. This breaks down the idea of the cleanroom that Yarn has tried to build, and is the main reason that the Quilt Mappings Team has decided to remove the cleanroom requirement.
+
+It's also clear that Mojang feels that their mappings should be used in this way - or at least have no wish to prevent it - given that they haven't intervened in other projects that use it in a similar way.
+
+It's become abundantly clear that Yarn's cleanroom isn't very clean, and that it doesn't serve any real purpose in today's modding ecosystem.
+
+### Changes to the Ecosystem
+
+As mentioned earlier, a lot has changed in the time since Yarn was concieved:
+
+* Most other modloader projects are using Mojmap, or something layered on top of it
+* MCP has been retroactively relicensed to the zlib license, and is no longer being maintained as an isolated mappings project
+* It's become clear that Microsoft has many other ways to kill off modloader projects, even though it doesn't have any discernable motivation to do so - and regardless of whether they'd be in the right to do so, it's unlikely that anyone would be able or willing to fund a court case in defense
+
```

Comment:
I believe we should add an "expected positive outcomes" section here, mentioning how allowing other mappings may improve cooperation between projects, let more people get help in the community channels, and allow mapping otherwise unmappable names.

Reply:
```suggestion

### Excpected Positive Outcomes

* Allowing other mappings in community spaces will allow people coming from other mod loaders to have a more welcoming experience as they don't have to learn a new mapping set just to recieve help
* Certain names could be improved and constants mapped by cross referencing names with Mojang Mappings, allowing for a stronger and more complete mapping of Minecraft
```
Anything else?

Reply:
How about: 
```suggestion
### Expected Positive Outcomes

The current rules needed to maintain the cleanroom have to be rather stringent in order to appear effective. We expect that dropping them could benefit the Quilt project in a few ways:

* Improved cooperation between modding projects - as other modloaders use different mappings, contributors who work on more than just Quilt-related projects would be considered "tainted" under the current rules. Dropping the cleanroom could increase the number of contributors both coming from other projects to help on Quilt, and coming from Quilt to collaborate on more ambitious cross-loader initiatives.
* Improved help in community spaces - allowing other mappings will allow developers coming from other mod loaders to have a more welcoming experience as they would not have to learn a new mapping set just to receive help.
* Improved mappings - some code constructs can be currently difficult to name without knowing the developers' intent. This challenge has been heightened with the release of the fields behind Minecraft's inlined constants. Being able to cross-reference names with Mojang Mappings could improve the quality and coverage of Quilt's own mappings when it comes to said constructs.
```